.github/workflows/master_read_codeowners.yml

```diff
@@ -0,0 +1,73 @@
+---
+name: Check Changed Files
+
+on:
+  workflow_call:
+    inputs:
+      changed-files:
+        description: 'All the changed files in PR'
+        required: true
+        type: string
+    outputs:
+      required-teams:
+        description: 'Teams required for approval'
+        value: ${{ jobs.read-codeowners.outputs.required-teams }}
+
+jobs:
+  
+  read-codeowners:
+    runs-on: non-build
+    outputs:
+      required-teams: ${{steps.extract-teams.outputs.teams}}
+    steps: 
+      - uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.CIPIPELINE_NEXUS_PASSWORD  }}
+          fetch-depth: 0
+          ref: ${{github.base_ref}}
+ 
+      - name: Extract team names from CODEOWNERS
+        id: extract-teams
+        env:
+          changed-files: "${{ inputs.changed-files }}"
+        run: |
+          python3 - <<EOF
+          import os
+          import sys
+          def extract_team_names(team_names):
+            return [team.replace('@TessellDevelopment/', '') for team in team_names]
```

Comment:
Use `GITHUB_REPOSITORY_OWNER` env variable (available by default) instead of hardcoding `TessellDevelopment`; Update in all files wherever this comment holds good